File: remote/src/main/java/com/kelsos/mbrc/mappers/GenreMapper.java
package com.kelsos.mbrc.mappers;

import android.support.annotation.NonNull;
import com.kelsos.mbrc.dao.GenreDao;
import com.kelsos.mbrc.domain.Genre;
import com.kelsos.mbrc.dto.library.GenreDto;
import java.util.List;
import rx.Observable;

public class GenreMapper {

  public static List<GenreDao> map(List<GenreDto> data) {
    return Observable.from(data).map(GenreMapper::map).toList().toBlocking().first();
  }

  public static GenreDao map(GenreDto dto) {
    GenreDao dao = new GenreDao();
    dao.setName(dto.getName());
    dao.setId(dto.getId());
    dao.setDateDeleted(dto.getDateDeleted());
    dao.setDateUpdated(dto.getDateUpdated());
    dao.setDateAdded(dto.getDateAdded());
    return dao;
  }

  public static List<Genre> mapToModel(List<GenreDao> genres) {
    return Observable.from(genres).map(GenreMapper::mapToModel).toList().toBlocking().first();
  }

  @NonNull public static Genre mapToModel(GenreDao dao) {
    return new Genre(dao.getId(), dao.getName());
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/mappers/GenreMapper.kt
package com.kelsos.mbrc.mappers

import com.kelsos.mbrc.dao.GenreDao
import com.kelsos.mbrc.domain.Genre
import com.kelsos.mbrc.dto.library.GenreDto

object GenreMapper {

  fun map(data: List<GenreDto>): List<GenreDao> {
    return data.map { map(it) }.toList()
  }

  fun map(dto: GenreDto): GenreDao {
    val dao = GenreDao()
    dao.name = dto.name
    dao.id = dto.id
    dao.dateDeleted = dto.dateDeleted
    dao.dateUpdated = dto.dateUpdated
    dao.dateAdded = dto.dateAdded
    return dao
  }

  fun mapToModel(genres: List<GenreDao>): List<Genre> {
    return genres.map { mapToModel(it) }.toList()
  }

  fun mapToModel(dao: GenreDao): Genre {
    return Genre(dao.id, dao.name)
  }
}
